toolchain/check/impl_lookup.cpp

```diff
@@ -266,6 +268,46 @@ static auto GetWitnessIdForImpl(Context& context, SemIR::LocId loc_id,
       context.sem_ir(), specific_id, impl.witness_id));
 }
 
+// In the case where `facet_const_id` is a facet, see if its facet type requires
+// that `specific_interface` is implemented. If so, return the witness from the
+// facet.
+static auto FindWitnessInFacet(Context& context, SemIR::LocId loc_id,
+                               SemIR::ConstantId facet_const_id,
+                               SemIR::SpecificInterface& specific_interface)
+    -> SemIR::InstId {
+  SemIR::InstId facet_inst_id =
+      context.constant_values().GetInstId(facet_const_id);
+  // FIXME: Should we convert from a FacetAccessType to its facet here?
+  SemIR::Inst facet_inst = context.insts().Get(facet_inst_id);
```

Comment:
nit: i'd maybe consider calling .type_id() on this immediate and skip the `facet_inst` local variable

Reply:
Done.